lib/ast.js

```diff
@@ -146,15 +168,22 @@ function opGen(opType, arity) {
 }
 
 // A reducing function for binary operators - use with [].reduce
-function leftAssociateGen(opType) {
+// initialValue's in array are ignored (or returned for empty array).
+function leftAssociateGen(opType, initialValue) {
   function reducer(result, current) {
     if (result === undefined) {
```

Comment:
Is this undefined check necessary. Seems like it could be hiding hard to debug bugs down the road.